posts/inside-rust/2024-01-02-this-development-cycle-in-cargo-1-76.md

```diff
@@ -0,0 +1,573 @@
+[RFC #3493](https://github.com/rust-lang/rfcs/pull/3493) changes cargo's dependency resolver so that users can opt-in to a pre-release in their `Cargo.lock`.
+This works well if users want to test for regressions in a pre-release dependency or want access to functionality early but don't require it (e.g. performance improvements).
+If a package requires something from a pre-release, like a new API, that should instead be specified in the version requirement which
+is more of the focus of [RFC #3263](https://github.com/rust-lang/rfcs/pull/3263).
+
+<!-- source: 2023-12-12 meeting -->
+Since we were already discussing postponing #3263,
+we discussed whether we should also postpone #3493.
+While we want to improve pre-releases,
+no one on the team is available to help shepherd this and
+the proposal would involve an invasive change to cargo that would likely be brittle.
+For how much time we do take to address pre-release,
+it was unclear if this was the most important.
+
+As we discussed it,
+we realized there was a solid precedence to base the design off of,
+[yanked packages](https://doc.rust-lang.org/cargo/commands/cargo-yank.html).
+We could also minimize risk by suggesting that the
+[`semver` package](https://crates.io/crates/semver)
+keep the existing version matching logic and expose the new behavior under a different function name.
+We also realized that this RFC is a prerequisite for RFC #3263 so that cargo could correctly unify pre-release and regular release version requirements.
+
+For bookkeeping purposes,
+there was a concern this would be require `Cargo.lock` changes.
+If so, then it would likely be complex enough that we'd need an experimental implementation first.
+However, we found `Cargo.lock` changes are unlikely to be needed after some further discussion.
+
+We updated the RFC and this is now waiting on author to wrap up the discussion.
+
+##### `cargo update --precise <yanked>`
+
+<!-- source: 2023-12-19 meeting -->
+Between [RFC #3493](https://github.com/rust-lang/rfcs/pull/3493) letting users force pre-releases through `--precise` and
+[rust-lang/cargo#12425](https://github.com/rust-lang/cargo/issues/12425) doing the same for breaking changes,
+it seemed fitting to extend this concept to yanked packages,
+resolving [rust-lang/cargo#4225](https://github.com/rust-lang/cargo/issues/4225).
+We felt we need to trust users in these scenarios.
+Users might have valid reasons to access yanked packages,
+whether its short-term to test something out
+or long term and they accept the risks.
+We considered an additional flag for this but pre-release and breaking changes don't need a flag.
+For breaking changes, the flag would be for use outside `--precise` for opting in for all packages.
+There is the possibility we'd want to extend that concept to pre-release and yanked packages.
+For a prepondance of caution, we reached out to a prior cargo team member in case there was context we were missing and they gave the greenlight.
```

Comment:
```suggestion
For a preponderance of caution, we reached out to a prior cargo team member in case there was context we were missing and they gave the greenlight.
```